src/LibChorusTests/VcsDrivers/Mercurial/HgVersionBranchingTests.cs

```diff
@@ -127,5 +127,45 @@ public void DoesNewBranchExist_No()
 				Assert.IsFalse(result, "The only branch should be default.");
 			}
 		}
+
+		[Test]
+		public void CanCreateVersionNumberBranch_BackwardCompatibilityTest()
+		{
+			// Setup
+			using(var repoWithFiles = RepositoryWithFilesSetup.CreateWithLiftFile(stestUser))
+			{
+				var integerBranchName = "70000068";
+				repoWithFiles.Synchronizer.SynchronizerAdjunct = new ProgrammableSynchronizerAdjunct(integerBranchName);
+				var branchingHelper = repoWithFiles.Repository.BranchingHelper;
+				Assert.AreEqual(1, branchingHelper.GetBranches().Count(),
+								"Setup problem in test, should be starting with one branch.");
+				// Make a new branch (should technically be on the remote with a different user...)
+				repoWithFiles.ReplaceSomething("nottheoriginal");
+				repoWithFiles.SyncWithOptions(new SyncOptions
+				{
+					DoPullFromOthers = false,
+					CheckinDescription = "version number branch",
+					DoSendToOthers = false
+				});
+
+				const string myVersion = ""; // Hg default branch name
+
+				// SUT
```

Comment:
It's not obvious that this is the SUT. Isn't the point of this test mainly that the SyncWithOptions is actually able to make a new branch with an integer name?